packages/hydrogen/src/customer/types.ts

```diff
@@ -0,0 +1,139 @@
+  ClientReturn,
+  ClientVariablesInRestParams,
+} from '@shopify/hydrogen-codegen';
+
+import type {CrossRuntimeRequest} from '../utils/request';
+
+import type {HydrogenSession} from '../hydrogen';
+
+// Return type of unauthorizedHandler = Return type of loader/action function
+// This type is not exported https://github.com/remix-run/react-router/blob/main/packages/router/utils.ts#L167
+type DataFunctionValue = Response | NonNullable<unknown> | null;
+
+type CustomerAPIResponse<ReturnType> = {
+  data: ReturnType;
+  errors: Array<{
+    message: string;
+    locations?: Array<{line: number; column: number}>;
+    path?: Array<string>;
+    extensions: {code: string};
+  }>;
+  extensions: {
+    cost: {
+      requestQueryCost: number;
+      actualQueryCakes: number;
+      throttleStatus: {
+        maximumAvailable: number;
+        currentAvailable: number;
+        restoreRate: number;
+      };
+    };
+  };
+};
+
+export interface CustomerAccountQueries {
+  // Example of how a generated query type looks like:
+  // '#graphql query q1 {...}': {return: Q1Query; variables: Q1QueryVariables};
+}
+
+export interface CustomerAccountMutations {
+  // Example of how a generated mutation type looks like:
+  // '#graphql mutation m1 {...}': {return: M1Mutation; variables: M1MutationVariables};
+}
+
+export type CustomerClient = {
+  /** Start the OAuth login flow. This function should be called and returned from a Remix action. It redirects the user to a login domain. An optional `redirectPath` parameter defines the final path the user lands on at the end of the oAuth flow. The default redirectPath is the page that ran handleUnauthorized, query or mutate with unauthorized customer, then `/account`. */
+  login: (redirectPath?: string) => Promise<Response>;
+  /** On successful login, the user redirects back to your app. This function validates the OAuth response and exchanges the authorization code for an access token and refresh token. It also persists the tokens on your session. This function should be called and returned from the Remix loader configured as the redirect URI within the Customer Account API settings. */
+  authorize: () => Promise<Response>;
+  /** Returns if the user is logged in. It also checks if the access token is expired and refreshes it if needed. */
+  isLoggedIn: () => Promise<boolean>;
+  /** Check for unauthorized customer and perform `unauthorizedHandler`. */
+  handleUnauthorized: () => void | DataFunctionValue;
+  /** Returns CustomerAccessToken if the user is logged in. It also run a expirey check and does a token refresh if needed. */
+  getAccessToken: () => Promise<string | undefined>;
+  /** Logout the user by clearing the session and redirecting to the login domain. It should be called and returned from a Remix action. */
+  logout: () => Promise<Response>;
+  /** Execute a GraphQL query against the Customer Account API. Usually you should first check if the user is logged in before querying the API. */
```

Comment:
Is the recommendation to check if the user is logged in still relevant?

Reply:
No, I will remove that.